### PR TITLE
[GAL-464] Implement Badges UI on Profile username

### PR DESCRIPTION
**Pull request by jakzaizzat**
**Changes**

Show art gobbler badge if the user has badge with `image URL`. 

In this case, we mock with ENS collection. 

**Demo on Profile**

https://user-images.githubusercontent.com/4480258/192439948-df90f0a9-e86c-4807-bb52-be9ffaa840ea.mp4

### CODE REVIEW

File: src/components/Badge/Badge.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  const handleMouseEnter = useCallback(() => {
+    setShowTooltip(true);
+  }, []);
+
+  const handleMouseExit = useCallback(() => {
+    setShowTooltip(false);
+  }, []);
+
+  // TODO: replace with badge contract address
+  const communityUrl = '/community/0x31982936afe7fc432ec1a2766bdba4d07a779b30';
```
Comment (jakzaizzat) [inline src/components/Badge/Badge.tsx:22]: @Robinnnnn What's the best way to get contract address art gobblers for community page?

Seems like their collection used opensea storefront - https://opensea.io/collection/art-gobbler

File: src/components/Badge/Badge.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    <InteractiveLink to={communityUrl}>
+      <StyledTooltip text={name} showTooltip={showTooltip} />
+      <StyledBadge src={imageURL} onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseExit} />
+    </InteractiveLink>
+  );
+}
+
+const StyledTooltip = styled(Tooltip)<{ showTooltip: boolean }>`
+  opacity: ${({ showTooltip }) => (showTooltip ? 1 : 0)};
+  transform: translateY(${({ showTooltip }) => (showTooltip ? -28 : 0)}px);
```
Comment (Robinnnnn) [inline src/components/Badge/Badge.tsx:34]: could we make the translation a little bit less intense? maybe something like `-28 : -24`

File: src/components/Badge/Badge.tsx (hunk 1 above)
Comment (Robinnnnn) [reply src/components/Badge/Badge.tsx:22]: hmm good question, so i think the badge should come with the contract address for us to link to it. this means we'll need to update the backend schema to return the contract:

image

^ this just returns an ID, but we the schema should really be `contract: Contract`. will ask someone to implement this tomorrow.

Comment (Robinnnnn) [reply src/components/Badge/Badge.tsx:22]: in the meantime, best practice graphql patterns is to pull the badge data as far down the react tree as possible – so `{ name, imageURL }: Props` should really be `{ badgeRef }: Props`

more expanded:
```ts
type Props = {
    // this type will be auto-generated by Relay once you hit "save".
    badgeRef: BadgeFragment$key 
}

export default function Badge({ badgeRef }: Props) {
    const badge = useFragment(
        graphql`
            fragment BadgeFragment on Badge {
                name
                imageURL
                contract {
                    address
                }
            }
        `,
        badgeRef
    )

    const { name, imageURL, contract: { address }} = badge;

    ...
}

// parent component
const user = useFragment(
    graphql`
      fragment UserGalleryHeaderFragment on GalleryUser {
        username @required(action: THROW)
        dbid
        bio
        badges {
          name
          ...BadgeFragment <-- main change
        }
      }
    `,
    userRef
  );

  ...

  {user.badges.map(badge => <Badge key={badge.name} badgeRef={badge} />)}

```

File: src/components/Badge/Badge.tsx (hunk 2 above)
Comment (jakzaizzat) [reply src/components/Badge/Badge.tsx]: Updated!

File: src/components/Badge/Badge.tsx (hunk 1 above)
Comment (jakzaizzat) [reply src/components/Badge/Badge.tsx]: TIL. Thanks a lot @Robinnnnn for the detail explanation!

**Comment by jakzaizzat** [comment]
@Robinnnnn I've updated the graphql schema and included in the badge community url

**Review by Robinnnnn — APPROVED** [review]
🔥🔥🔥🔥🔥

thank you jakz! so stokeddd!

**Pull request merged**